tests/behaviour/steps/lib.rs

```diff
@@ -86,20 +90,43 @@ impl Context {
             .execution_has_failed()
     }
 
-    async fn after_scenario(&mut self) -> Result<(), ()> {
+    async fn after_scenario(&mut self, clean_databases: bool) -> Result<(), ()> {
+        if let Some(_) = &self.active_transaction {
+            self.close_transaction()
+        }
+
+        if clean_databases {
+            let database_names = self.database_names();
+            for database_name in database_names {
+                self.server_mut().unwrap().lock().unwrap().delete_database(database_name).unwrap();
+            }
+        }
+
         Ok(())
     }
 
-    pub fn server(&self) -> Option<&typedb::Server> {
+    pub fn close_transaction(&mut self) {
+        match self.take_transaction().unwrap() {
+            ActiveTransaction::Read(tx) => tx.close(),
+            ActiveTransaction::Write(tx) => tx.close(),
+            ActiveTransaction::Schema(tx) => tx.close(),
+        }
+    }
+
+    pub fn server(&self) -> Option<&Arc<Mutex<typedb::Server>>> {
         self.server.as_ref()
     }
 
-    pub fn server_mut(&mut self) -> Option<&mut typedb::Server> {
+    pub fn server_mut(&mut self) -> Option<&mut Arc<Mutex<typedb::Server>>> {
         self.server.as_mut()
     }
 
-    pub fn databases(&self) -> &HashMap<String, Arc<Database<WALClient>>> {
-        self.server().unwrap().databases()
+    pub fn databases(&self) -> HashMap<String, Arc<Database<WALClient>>> {
+        self.server().unwrap().lock().unwrap().databases().clone()
+    }
+
+    pub fn database_names(&self) -> HashSet<String> {
+        self.server().unwrap().lock().unwrap().databases().keys().map(|name| name.to_owned()).collect()
```

Comment:
I have a separate function not increase the `Arc` counter while getting `databases()` for the cleanup